spock-specs/src/test/groovy/org/spockframework/smoke/extension/TempDirExtensionSpec.groovy

```diff
@@ -0,0 +1,104 @@
+package org.spockframework.smoke.extension
+
+import org.spockframework.util.Beta
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Stepwise
+import spock.lang.TempDir
+
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.attribute.PosixFilePermissions
+
+/**
+ * @author dqyuan
+ */
+@Beta
+@Stepwise
+class TempDirExtensionSpec extends Specification {
+
+  @Shared
+  @TempDir
+  Path sharedDir
+  static Path preShared
```

Comment:
Maybe write out `previousShared` and `previousIteration`?
Imho `preShared` sounds like "before it was shared" not "what was shared before". :-)